ccm_web/client/src/pages/Home.tsx

```diff
@@ -33,7 +36,38 @@ interface HomeProps extends LtiProps {
 }
 
 function Home (props: HomeProps): JSX.Element {
+  const { enqueueSnackbar } = useSnackbar()
   const classes = useStyles()
+  const [course, setCourse] = useState<undefined|CanvasCourseBase>(undefined)
+
+  const [doLoadCourse, isLoadingCourse, getCourseError] = usePromise(
+    async () => await getCourse(props.ltiKey, props.globals.course.id),
+    (value: CanvasCourseBase) => setCourse(value)
+  )
+
+  useEffect(() => {
+    void doLoadCourse()
+  }, [])
+
+  const setCourseNameAsync = async (
+    newCourseName: string
+  ): Promise<CanvasCourseBase|undefined> => {
+    return await apiSetCourseName(props.ltiKey, props.globals.course.id, newCourseName)
+  }
+
+  const [doSetCourseName, setCourseNameLoading, setCourseNameError] = usePromise<CanvasCourseBase|undefined, typeof setCourseNameAsync>(
+    setCourseNameAsync,
+    (course: CanvasCourseBase|undefined) => {
+      setCourse(course)
+      enqueueSnackbar('Course name saved', { variant: 'success' })
+    }
+  )
+
+  useEffect(() => {
+    if (setCourseNameError !== undefined) {
+      enqueueSnackbar('Error saving course name: ' + setCourseNameError.message, { variant: 'error' })
```

Comment:
Error message from backend being spit out here verbatim.  Probably not the best idea.

Reply:
I trigged the error case with unknown canvas endpoint and that will throw 422 error and currently the error message the BE is sending is HTml document. When I changed to get the correct response variable then it will show as below
![Screen Shot 2021-07-01 at 1 52 10 PM](https://user-images.githubusercontent.com/8579775/124168819-9d4b8b00-da73-11eb-9052-909cedf63bfe.png)

I don't know if user want more info here other than `course name was not changed` 

You can do `Error saying course name due to setCourseNameError.message, { variant: 'error' }`, what do you think?

Reply:
I wasn't sure what to do here.  I think to really handle it well there needs to be more to the Error object.. perhaps some custom errors that extend  the Error class.  I know @ssciolla was trying to provide error feedback to the user in the case of an invalid course name, so in that case it would make sense to give more details, while in general I believe its pretty much always a bad idea to pass errors along verbatim.  I'll see if i can do some custom error type.

Reply:
Yeah I'm not sure what do do here besides say the save failed.  It's not going to give any helpful feedback when its a validation sort of error, though.  More than that and its going to complicate things quite a bit I think.